### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dissom**

### CODE REVIEW

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
try to use only one return in def wrapper

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> None:
```
Comment (vsmutok) [inline app/main.py:9]: add functools.wraps

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,16 @@
...
+        key = (frozenset(args), frozenset(kwargs.items()))
+        if key not in cache_dict:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cache_dict[key] = result
+        else:
+            print("Getting from cache")
+            result = cache_dict[key]
+        return result
```
Comment (vsmutok) [inline app/main.py:18]:
```suggestion
            cache_dict[key] = func(*args, **kwargs)
        else:
            print("Getting from cache")
        return cache_dict[key]
```

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vsmutok — APPROVED** [review]
well done!

**Pull request closed without merging**